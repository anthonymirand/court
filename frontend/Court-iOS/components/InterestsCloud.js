import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'expo';

import { ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Colors from '../constants/Colors';

import InterestsItem from '../components/InterestsItem';

/**
* Displays a 'cloud' of interests in a flex box
*/
export default class InterestsCloud extends React.Component {
  render() {
    const { color, interests, onAddInterest, onRemoveInterest, recommendations } = this.props;
    console.log(recommendations);
    return (
      // Scroll View encompasing the results
      <View style={styles.outerView}>
        {recommendations.map(function(item, index) {
          return (
            <View key={index}>
              <InterestsItem
                id={item.id}
                color={color}
                showRemoveIcon={false}
                selected={!interests || item.id in interests}
                title={item.title}
                description={item.description}
                imgUrl={item.imgUrl}
                onAddInterest={onAddInterest}
                onRemoveInterest={onRemoveInterest}
              />
            </View>
          );
        })}
      </View>
    );
  }
}

InterestsCloud.propTypes = {
  /**
  * Object containing the user's current selected interests
  */
  interests: PropTypes.object,
  /**
  * Callback function for when a user selects a new interest
  */
  onAddInterest: PropTypes.func,
  /**
  * Callback function for when a user removes an interest
  */
  onRemoveInterest: PropTypes.func,
  /**
  * List of recommended interests given a specific search
  */
  recommendations: PropTypes.arrayOf(PropTypes.object).isRequired,
}

const styles = StyleSheet.create({
  outerView: {
    flexDirection: 'row',
    justifyContent: 'center',
    flexWrap: 'wrap',
    paddingTop: 15,
  },
});
