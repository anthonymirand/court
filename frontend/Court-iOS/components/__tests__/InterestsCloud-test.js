import 'react-native';
import React from 'react';
import InterestsCloud from '../InterestsCloud';
import renderer from 'react-test-renderer';

it('Renders Default InterestsCloud', () => {
  const mocks = { onAddInterest: jest.fn(), onRemoveInterest: jest.fn() };
  const interests = {'kgs://12hv3': {title: 'The Office', description: 'American Sitcom', imgUrl: 'test.url'}};
  const recommendations = [{id: 'kgs://jsh12', title: 'Interest Rec 1'}, {id: 'kgs://3jh43', title: 'Interest Rec 2'}];
  const tree = renderer.create(
    <InterestsCloud
      interests={interests}
      recommendations={recommendations}
      onAddInterest={mocks.onAddInterest}
      onRemoveInterest={mocks.onRemoveInterest}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
