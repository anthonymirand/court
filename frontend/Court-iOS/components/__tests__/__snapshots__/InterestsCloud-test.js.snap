// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default InterestsCloud 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 5,
            "padding": 5,
          }
        }
      >
        // Optional: add image preview here
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "column",
              "marginLeft": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            Interest Rec 1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 5,
            "padding": 5,
          }
        }
      >
        // Optional: add image preview here
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "column",
              "marginLeft": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            Interest Rec 2
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
